exchanges/fee/fee.go

```diff
@@ -0,0 +1,74 @@
+package fee
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/thrasher-corp/gocryptotrader/common/key"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+var (
+	// ErrExchangeNameEmpty is an error for when the exchange name is empty
+	ErrExchangeNameEmpty = errors.New("exchange name is empty")
+	// ErrFeeRateNotFound is an error for when the fee rate is not found
+	ErrFeeRateNotFound = errors.New("fee rate not found")
+	exchangeFees       = Fees{all: make(map[key.ExchangePairAsset]Rates)}
+)
+
+// Fees holds the fees for each exchange.
+type Fees struct {
+	all map[key.ExchangePairAsset]Rates
```

Comment:
I think the main thing with this is that I feel it might be good as a subdirectory to "account". Most fees are going to be tied to user accounts, we can pass credentials via context and we can therefore store different fees for different accounts on the same `key.ExchnagePairAsset`s.

For me locally, I have an extra key:
```go

type ExchangePairAssetCredentials struct {
	Exchange    string
	Asset       asset.Item
	Base        *currency.Item
	Quote       *currency.Item
	Credentials *credentials.Credentials
}

func (k *ExchangePairAssetCredentials) KeyNoCreds() ExchangePairAsset {
	return ExchangePairAsset{
		Exchange: k.Exchange,
		Base:     k.Base,
		Quote:    k.Quote,
		Asset:    k.Asset,
	}
}

func (k *ExchangePairAssetCredentials) Pair() currency.Pair {
	if k == nil || (k.Base == nil && k.Quote == nil) {
		return currency.EMPTYPAIR
	}
	return currency.NewPair(k.Base.Currency(), k.Quote.Currency())
}
```
Just means on the wrapper implementation, we grab the credentials from the passed in context

edit: Just saw your todo, but I think it should be done for this PR 😄